src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx

```diff
@@ -0,0 +1,196 @@
+      },
+      {
+        onSuccess: () => {
+          onBack?.()
+        },
+      }
+    )
+  }
+  const nameError = (): string | undefined => {
+    if (name.length > 60) {
+      return t.overCharacterLimitError
+    }
+    return undefined
+  }
+
+  const codeError = (): string | undefined => {
+    // Only check code on any input
+    if (code.trim().length === 0) {
+      return undefined
+    }
+
+    // If a non-digit exists
+    if (/\D/.test(code)) {
+      return 'Number only.'
+    }
+
+    if (code.length < minCodeLength || code.length > maxCodeLength) {
+      return 'Code must be between 4, and 8 digits.'
+    }
+
+    return undefined
+  }
+
+  const generateCode = (): void => {
+    // Randomize code length
+    const length = getRandomInt({ min: minCodeLength, max: maxCodeLength })
+    // Fill each digit with a random int from 0-9
+    const generated = Array.from({ length }, () =>
+      getRandomInt({ min: 0, max: 9 })
+    ).join('')
+
+    setCode(generated)
+  }
+
+  const canSave = (): boolean => {
+    return (
+      name.trim().length > 0 &&
+      nameError() === undefined &&
+      code.trim().length > 0 &&
+      codeError() === undefined &&
+      !createAccessCode.isLoading
+    )
+  }
+
+  return (
+    <>
+      <ContentHeader
+        title={t.addNewAccessCode}
+        subheading={device.properties.name}
+        onBack={onBack}
+      />
+      <div className='seam-main'>
+        <FormField>
+          <InputLabel>{t.nameInputLabel}</InputLabel>
+          <TextField
+            size='large'
+            clearable
+            value={name}
+            onChange={setName}
+            hasError={nameError() != null}
+            helperText={nameError()}
+          />
+        </FormField>
+
+        <FormField className='seam-code-field'>
+          <InputLabel>{t.codeInputLabel}</InputLabel>
+          <TextField
+            size='large'
+            clearable
+            value={code}
+            onChange={setCode}
+            onFocus={toggleCodeInputFocused}
+            onBlur={toggleCodeInputFocused}
+            hasError={codeError() != null}
+          />
+          <div
+            className={classNames('seam-bottom', {
+              'seam-visible': codeInputFocused,
+            })}
+          >
+            <ul
+              className={classNames('seam-requirements', {
+                'seam-error': codeError() != null,
+              })}
+            >
+              <li>{t.codeRequirementLength}</li>
+              <li>{t.codeRequirementNumbersOnly}</li>
+            </ul>
+            <Button
+              size='small'
+              onMouseDown={(e) => {
+                e.preventDefault() // Disable stealing input focus
+                generateCode()
+              }}
+            >
+              {t.codeGenerateButton}
+            </Button>
+          </div>
+        </FormField>
+        <div className='seam-actions'>
+          <Button onClick={onBack}>Cancel</Button>
```

Comment:
```suggestion
          <Button onClick={onBack}>{t.cancel}</Button>
```

Reply:
Updated in https://github.com/seamapi/react/pull/271/commits/96d206465f6498c2c5b8a25bcf168a2addb62f9a